modules/programs/pianobar.nix

```
@@ -0,0 +1,118 @@
+/**
+  ## Minimal configurations to use this module
+
+  ```nix
+  { ... }:
+  {
+    programs.pianobar = {
+      enable = true;
+      user = "groovy-tunes@example.com";
+      password_command = "cat /run/secrets/pianobar/groovy-tunes";
+    };
+  }
+  ```
+
+  > Note it is recommended to use `sops-nix`, or similar, secrets management
+  > solution for providing `programs.pianobar.password_command` value.
+*/
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    literalExpression
+    ;
+
+  inherit (lib.types)
+    nonEmptyStr
+    submodule
+    ;
+
+  inherit (pkgs.formats)
+    keyValue
+    ;
+
+  keyValueFromat = keyValue { };
+
+  cfg = config.programs.pianobar;
+in
+{
+  meta.maintainers = [
+    # lib.maintainers.S0AndS0
+    ## TODO: Trade above for below when `node.nixpkgs.locked.rev` is at or beyond
+    ##       4d48a4e93b9ffbd291b2d9ca3315848e27eed800
+    {
+      name = "S0AndS0";
+      email = "S0AndS0@digital-mercenaries.com";
+      github = "S0AndS0";
+      githubId = 4116150;
+      matrix = "@s0ands0:matrix.org";
+    }
+  ];
+
+  options.programs.pianobar = {
+    enable = mkEnableOption "Enable pianobar";
+
+    package = mkPackageOption pkgs "pianobar" {
+      default = [ "pianobar" ];
```

```suggestion
      nullable = true;
```

The string argument will actually set the default for us. But, we can offer config only support by allowing the package to be null.